src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -0,0 +1,902 @@
+        if from_row > 0:
+            request["start"] = from_row
+        if max_rows >= 0:
+            request["rows"] = max_rows
+        if add_sort and self._sortcriteria != {}:
+            request["sort"] = [self._sortcriteria]
+        return request
+
+    def _perform_query(self, from_row=0, max_rows=-1):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 0).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                yield item
+                current += 1
+                numrows += 1
+
+                if max_rows > 0 and numrows == max_rows:
+                    still_querying = False
+                    break
+
+            if current >= self._total_results:
+                break
+
+    def _run_async_query(self, context):
+        """
+        Executed in the background to run an asynchronous query.
+
+        Args:
+            context (object): Not used, always None.
+
+        Returns:
+            list: Result of the async query, which is then returned by the future.
+        """
+        url = self._build_url("")
+        request = self._build_request(0, -1)
+        resp = self._cb.post_object(url, body=request)
+        result = resp.json()
+        self._total_results = result["num_found"]
+        self._count_valid = True
+        results = result.get("results", [])
+        return [item for item in results]
+
+    def _count(self):
+        """
+        Returns the number of results from the run of this query.
+
+        Returns:
+            int: The number of results from the run of this query.
+        """
+        if self._count_valid:
+            return self._total_results
+        url = self._build_url("")
+        request = self._build_request(0, -1)
+        resp = self._cb.post_object(url, body=request)
+        result = resp.json()
+
+        self._total_results = result["num_found"]
+        self._count_valid = True
+
+        return self._total_results
+
+
+class DeviceVulnerabilityQuery(VulnerabilityQuery):
+    """Query Class for the DeviceVulnerability"""
+
+    def __init__(self, doc_class, cb):
+        """
+        Initialize the DeviceVulnerabilityQuery.
+
+        Args:
+            doc_class (class): The model class that will be returned by this query.
```

Comment:
This needs to be renamed as you are not going to return a Device which is what you are receiving here instead you will be returning DeviceVulnerability. I think it's fine to hard code the `self.doc_class = DeviceVulnerability` so you should rename the parameter to device and store this into a separate variable `self.device` or something